smtpd/mail.mboxfile.c

```diff
@@ -71,10 +82,19 @@ mboxfile_engine(const char *sender, const char *filename)
 
 	time(&now);
 
-	fd = open(filename, O_CREAT | O_APPEND | O_WRONLY | O_EXLOCK, 0600);
+#ifdef HAVE_DECL_O_EXCL
```

Comment:
we avoid ifdef as much as possible in smtpd, i suggest we discuss this separately from the other diffs which are mostly ok minus minor fixes.

Reply:
I can find/make a macro solution for this if you like.

Reply:
Or maybe a function in the openbsd compat part, such as `open_locked_atomic()` or something.